addons/mail/models/discuss/discuss_channel.py

```diff
@@ -522,13 +522,12 @@ def _add_members(
         all_new_members = self.env["discuss.channel.member"]
         for channel in self:
             members_to_create = []
-            existing_members = self.env['discuss.channel.member'].search(expression.AND([
-                [('channel_id', '=', channel.id)],
-                expression.OR([
-                    [('partner_id', 'in', partners.ids)],
-                    [('guest_id', 'in', guests.ids)]
-                ])
-            ]))
+            existing_members = self.env['discuss.channel.member'].search(
+                Domain('channel_id', '=', channel.id) & (
+                    Domain('partner_id', 'in', partners.ids)
+                    | Domain('guest_id', 'in', guests.ids)
+                )
+            )
```

Comment:
I would format it this way instead:
```suggestion
            existing_members = self.env["discuss.channel.member"].search(
                Domain("channel_id", "=", channel.id)
                & (Domain("partner_id", "in", partners.ids) | Domain("guest_id", "in", guests.ids))
            )
```